src/grasshopper/lib/fixtures/__init__.py

```diff
@@ -531,3 +564,87 @@ def type_check_list_of_strs(list_of_strs):
+
+
+def _get_composite_weighted_user_classes(
+    full_scenarios_list, composite_scenario_contents
+):
+    """Generate a dictionary of journey classes with their associated weights."""
+    weighted_user_classes = {}
+    child_scenario_specs = _get_child_scenario_specs(
+        full_scenarios_list, composite_scenario_contents
+    )
+    for child_scenario_spec in child_scenario_specs:
+        base_journey_class = _import_class_with_journey(
+            absolute_file_path=f"{os.getcwd()}/"
+            f'{child_scenario_spec.get("test_file_name")}'
+        )
+        base_class_name = base_journey_class.__name__
+        composite_class_name = (
+            f"composite_journey_class_{base_class_name}_{uuid.uuid4()}"
+        )
+
+        # dynamically create a child class which inherits from the base class,
+        # required for having separate scenario args
+        child_journey_class = type(
+            composite_class_name,
+            (base_journey_class,),
+            {
+                "_incoming_test_parameters": child_scenario_spec.get(
+                    "grasshopper_scenario_args"
+                )
+            },
+        )
+        weighted_user_classes[child_journey_class] = child_scenario_spec.get("weight")
+    return weighted_user_classes
+
+
+def _import_class_with_journey(absolute_file_path):
+    """Import and return a class with 'journey' in its name from a module file."""
+    try:
+        module_name = os.path.splitext(absolute_file_path)[0]
+        spec = importlib.util.spec_from_file_location(module_name, absolute_file_path)
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+
+        # Now, inspect the module's attributes to find the class with "journey" in
+        # its name
+        for name, obj in vars(module).items():
+            if (
+                isinstance(obj, type)
+                and "journey" in name.lower()
+                and "base" not in name.lower()
+            ):
+                return obj
+
+        # If no class with "journey" in its name is found, return None or raise an
+        # exception
+        raise ImportError("No class with 'journey' in its name")
+
+    except ImportError as e:
+        # Handle import errors, e.g., file not found
+        print(f"Import error: {e}")
+        return None
+
+
+def _get_child_scenario_specs(full_scenarios_list, composite_scenario_contents):
+    """Extract and prepare child scenario specs given composite scenario contents."""
+    child_scenarios = composite_scenario_contents.get("child_scenarios")
+    child_scenario_specs = []
+    for child_scenario in child_scenarios:
+        child_scenario_name = child_scenario.get("scenario_name")
+        child_scenario_overrides = child_scenario.get(
+            "grasshopper_scenario_arg_overrides", {}
+        )
+        child_scenario_spec = full_scenarios_list.get(child_scenario_name)
+        if child_scenario_spec is None:
```

Comment:
is there any logic to prevent the user from listed a composite scenario as a child scenario? or does that work? i think it probably does not, given how we've designed this.

Reply:
added some safeguards preventing recursive child scenarios